blocks/api/test/paste.js

```diff
@@ -45,3 +48,59 @@ describe( 'normaliseToBlockLevelNodes', () => {
 	} );
 } );
 
+describe( 'paste', () => {
+	beforeAll( () => {
+		registerBlockType( 'test/small', {
+			category: 'common',
+			attributes: {
+				content: {
+					type: 'array',
+					source: children( 'small' ),
+				},
+			},
+			transforms: {
+				from: [
+					{
+						type: 'raw',
+						isMatch: ( node ) => node.nodeName === 'SMALL',
+					},
+				],
+			},
+			save: () => {},
+		} );
+
+		registerBlockType( 'test/unknown', {
+			category: 'common',
+			attributes: {
+				content: {
+					type: 'string',
+					source: prop( 'innerHTML' ),
+				},
+			},
+			save: () => {},
+		} );
+
+		setUnknownTypeHandlerName( 'test/unknown' );
+	} );
+
+	it( 'should convert recognised pasted content', () => {
+		const pastedBlock = paste( createNodes( '<small>test</small>' ) )[ 0 ];
+		const block = createBlock( 'test/small', { content: [ 'test' ] } );
+
+		equal( pastedBlock.name, block.name );
+		deepEqual( pastedBlock.attributes, block.attributes );
+	} );
+
+	it( 'should handle unknown pasted content', () => {
+		const pastedBlock = paste( createNodes( '<big>test</big>' ) )[ 0 ];
+
+		equal( pastedBlock.name, 'test/unknown' );
+		equal( pastedBlock.attributes.content, '<big>test</big>' );
+	} );
+
+	afterAll( () => {
```

Comment:
Minor: Conventionally setup and teardown occurs together before any test cases.

https://facebook.github.io/jest/docs/setup-teardown.html